ResearchKit/Common/ORKAxisView.m

```diff
@@ -0,0 +1,135 @@
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE 
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL 
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, 
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+ */
+
+ 
+#import "ORKAxisView.h"
+
+
+@interface ORKAxisView ()
+
+@property (nonatomic, strong) NSMutableArray *titleLabels;
+
+@property (nonatomic) ORKGraphAxisType axisType;
+
+@end
+
+
+@implementation ORKAxisView
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        [self sharedInit];
+    }
+    return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    self = [super initWithFrame:frame];
+    if (self) {
+        [self sharedInit];
+    }
+    return self;
+}
+
+- (void)sharedInit {
+    _titleLabels = [NSMutableArray new];
+    _leftOffset = 0;
+}
+
+- (void)layoutSubviews {
+    CGFloat segmentWidth = CGRectGetWidth(self.bounds) / (self.titleLabels.count - 1);
+    CGFloat labelWidth = segmentWidth;
+    
+    CGFloat labelHeight = (self.axisType == ORKGraphAxisTypeX) ? CGRectGetHeight(self.bounds) * 0.77 : 20;
+    
+    for (NSUInteger i = 0; i < self.titleLabels.count; i++) {
+        
+        CGFloat positionX = (self.axisType == ORKGraphAxisTypeX) ? (self.leftOffset + i * segmentWidth) : 0;
+        
+        if (i == 0) {
+            // Shift the first label to accomodate the month text.
+            positionX -= self.leftOffset;
+        }
+        
+        UILabel *label = (UILabel *)self.titleLabels[i];
+        
+        if (label.text) {
+            labelWidth = [label.text boundingRectWithSize:CGSizeMake(CGFLOAT_MAX, labelHeight)
+                                                  options:(NSStringDrawingUsesFontLeading | NSStringDrawingUsesLineFragmentOrigin)
+                                               attributes:@{NSFontAttributeName:label.font}
+                                                  context:nil].size.width;
+            labelWidth = MAX(labelWidth, 15);
+            labelWidth += self.landscapeMode ? 14 : 8; //padding
+        }
+        
+        if (i == 0) {
+            label.frame  = CGRectMake(positionX, (CGRectGetHeight(self.bounds) - labelHeight) / 2, labelWidth, labelHeight);
+        } else {
+            label.frame  = CGRectMake(positionX - labelWidth / 2, (CGRectGetHeight(self.bounds) - labelHeight) / 2, labelWidth, labelHeight);
+        }
+        
+        if (i == self.titleLabels.count - 1) {
```

Comment:
This special visual style for the last label is nice, but feels like it belongs to the `setupLabels` method.